### PR TITLE
controllers: add CephBlockPool to StorageCluster relatedObjects

### PR DESCRIPTION
**Pull request by umangachapagain**
Adding default CephBlockPool to relatedObjects in StorageCluster
since it is an object of interest for managing various pool
level features.

Signed-off-by: Umanga Chapagain <chapagainumanga@gmail.com>

### CODE REVIEW

File: controllers/storagecluster/cephblockpools.go [hunk 1]
```diff
@@ -83,6 +86,29 @@ func (obj *ocsCephBlockPools) ensureCreated(r *StorageClusterReconciler, instanc
...
 		}
+
+		if !reflect.DeepEqual(existing, cephv1.CephBlockPool{}) {
+			// If CephBlockPool is not empty, add it to the list of RelatedObjects
+			objectRef, err := reference.GetReference(r.Scheme, &existing)
+			if err != nil {
+				r.Log.Error(err, "Unable to get CephBlockPool ObjectReference.", "CephBlockPool", klog.KRef(existing.Namespace, existing.Name))
+				return err
+			}
+			err = objectreferencesv1.SetObjectReference(&instance.Status.RelatedObjects, *objectRef)
```
Comment (nbalacha) [inline controllers/storagecluster/cephblockpools.go:97]: Should the reference be removed when the cephblockpool is deleted?

File: controllers/storagecluster/cephblockpools.go [hunk 2]
```diff
@@ -83,6 +86,29 @@ func (obj *ocsCephBlockPools) ensureCreated(r *StorageClusterReconciler, instanc
 				return err
 			}
 		}
+
+		if !reflect.DeepEqual(existing, cephv1.CephBlockPool{}) {
```
Comment (nbalacha) [inline controllers/storagecluster/cephblockpools.go:90]: Should this be done each time or only when the CephBlockPool is first created (switch case errors.IsNotFound)?

**Review by nbalacha — COMMENTED** [review]
A few questions

**Comment by umangachapagain** [comment]
Closing this as it is no longer required.

**Pull request closed without merging**